scripts/apps/highlights/services/HighlightsService.ts

```diff
@@ -112,17 +111,19 @@ export function HighlightsService(api, $q, $cacheFactory, packages: IPackagesSer
     };
 
     /**
-     * Mark an item for a highlight
+     * Mark/Unmark an item for a highlight
      */
-    service.markItem = function(highlight, markedItem) {
+    service.markItem = function(highlight: string, markedItem: IArticle) {
+        const addToHighlight = markedItem.highlights.includes(highlight) === false;
+
         return trackArticleActionProgress(
             () => api.save(
                 'marked_for_highlights',
                 {highlights: [highlight], marked_item: markedItem._id},
             ),
             markedItem._id,
-            gettext('Item marked'),
-            gettext('Couldn\'t mark item'),
+            gettext(addToHighlight ? 'Item marked' : 'Item unmarked'),
```

Comment:
you can't have ternaries or any other logic inside `gettext`. The only thing allowed is plus operator for concatenating strings. Otherwise, string extraction wouldn't work.

Use `addToHighlight ? gettext('Item marked') : gettext('Item unmarked')`